docs/rules/id-length.md

```diff
@@ -0,0 +1,77 @@
+# Limit minimum and maximum length for identifiers (id-length)
+
+Identifier names such as `e`, `x`, `_t` or sometimes very long ones like `hashGeneratorResultOutputContainerObject` which usually makes the code harder to read and less maintainable. To prevent this, some code bases may enforce a minimum and/or maximum identifier length. (usually min 2-chars)
+
+```js
+// id-length: 1  // minimum 2-chars
+var x = 5; // too short
+```
+
+## Rule Details
+
+This rule is aimed at increasing code readability and maintainability by enforcing an identifier length convention. It will warn on any type of identifier which doesn't conform to length limits (upper and lower). The rule ignores common prefixes and suffixes up to two chars of `_` and `$` in any combination.
+
+It has exceptions for commonly used single-char identifier names such as `i` or `$`. It also allows the programmers to silently by-pass this check by using "quoted" property names or calculated property access to allow potential server-side data requirements.
```

Comment:
@nzakas Would default exceptions be acceptable if specifying an exceptions list overrode the defaults, allowing users to remove the default exceptions? A couple examples:

```js
{
    "id-length": 2, // Exceptions default to ["i", "j", "_", "$"]
    "id-length": [2, {
        "exceptions": [] // No exceptions
    }],
    "id-length": [2, {
        "exceptions": ["$"] // Only allow "$"
    }],
    "id-length": [2, {
        "exceptions": ["i", "j", "_", "$", "x"] // Defaults plus "x"
    }]
}
```

I'm proposing this because I like those defaults, though I completely agree it should be possible to override them. (Conveniently, the modification to support this would only require changing the empty fallback `[]` to include the defaults.)